implementations/micrometer-registry-prometheus/src/main/java/io/micrometer/prometheusmetrics/PrometheusMeterRegistry.java

```diff
@@ -549,13 +558,8 @@ private void onMeterRemoved(Meter meter) {
         }
     }
 
-    private MetricMetadata getMetadata(Meter.Id id) {
-        return getMetadata(id, "");
-    }
-
-    private MetricMetadata getMetadata(Meter.Id id, String suffix) {
-        String name = config().namingConvention().name(id.getName(), id.getType(), id.getBaseUnit()) + suffix;
-        String help = prometheusConfig.descriptions() ? Optional.ofNullable(id.getDescription()).orElse(" ") : " ";
+    private MetricMetadata getMetadata(String name, @Nullable String description) {
+        String help = prometheusConfig.descriptions() ? Optional.ofNullable(description).orElse(" ") : " ";
```

Comment:
This was a copy-paste form the 0.x client code but eliminating the `Optional` might also have some positive impact on performance.

Reply:
Agreed and it stood out to me too when making these changes, but I wanted to keep the changes here focused to one issue. Let's take it up in a separate change.